projects/v3/src/app/services/achievement.service.ts

```diff
@@ -39,64 +45,126 @@ export class AchievementService {
   isPointsConfigured = false;
 
   constructor(
+    private apolloService: ApolloService,
     private request: RequestService,
     private utils: UtilsService,
-    private demo: DemoService,
-  ) { }
+    private demo: DemoService
+  ) {}
+
+  /**
+   * [graphQLGetAchievements description]
+   * @link https://intersective.github.io/core-graphql-api/query.doc.html#:~:text=achievements(filter%3A%20String%2C%20type%3A%20String%2C%20active%3A%20Boolean)%3A%20%5BAchievement%5D
+   * @param   {string}                     filter  keyword to search for
+   * @param   {string}                     type    achievement type
+   * @param   {boolean}                    active  status of is active?
+   *
+   * @return  {Observable<Achievement>[]}          achievement list in badges
+   */
+  graphQLGetAchievements(
+    filter?: string,
+    type?: string,
+    active?: boolean
+  ): Observable<Achievement[]> {
+    return this.apolloService
+      .graphQLFetch(
+        `query achievements($filter: String, $type: String, $active: Boolean) {
+          achievements(filter: $filter, type: $type, active: $active) {
+            id
+            name
+            description
+            type
+            badge
+            openBadge
+            points
+            isEarned
+            earnedDate
+            progress
+            active
+            certificateUrl
+          }
+        }`,
+        {
+          variables: {
+            filter,
+            type,
+            active,
+          },
+        }
+      )
+      .pipe(
+        map(
+          (res: {
+            data: {
+              badges: Achievement[];
+            };
+          }) => {
+            return res?.data?.badges || [];
```

Comment:
the response of this endpoint will be `res.data.achievements`. Please merge this code to trunk and test on stage first, before merging it to the golive branch